src/server/pfs/server/driver_v2.go

```diff
@@ -513,3 +515,55 @@ func (d *driverV2) compactionWorker() {
 	// Never ending backoff should prevent us from getting here.
 	panic(err)
 }
+
+func (d *driverV2) copyFile(pachClient *client.APIClient, src *pfs.File, dst *pfs.File, overwrite bool) (retErr error) {
+	if overwrite {
+		// TODO: after delete merging is sorted out add overwrite support
+		return errors.New("overwrite not yet supported")
+	}
+	return d.withFileSet(pachClient.Ctx(), dst.Commit.Repo.Name, dst.Commit.ID, func(dstFs *fileset.FileSet) error {
+		return d.getTarConditional(pachClient.Ctx(), src.Commit.Repo.Name, src.Commit.ID, src.Path, func(rf *FileReader) error {
+			r, w := io.Pipe()
+			eg := errgroup.Group{}
+			eg.Go(func() error {
+				return rf.Get(w)
+			})
+			eg.Go(func() error {
+				return dstFs.Put(r)
+			})
+			return eg.Wait()
+		})
+	})
+}
+
+func (d *driverV2) inspectFile(pachClient *client.APIClient, file *pfs.File) (fi *pfs.FileInfoV2, retErr error) {
+	commitInfo, err := d.inspectCommit(pachClient, file.Commit, pfs.CommitState_STARTED)
+	if err != nil {
+		return nil, err
+	}
+	commit := commitInfo.Commit
+	stopIter := errors.New("stop iteration")
+	path := "/" + strings.Trim(file.Path, "/")
+	err = d.getTarConditional(pachClient.Ctx(), commit.Repo.Name, commit.ID, path, func(fr *FileReader) error {
```

Comment:
It might make sense to just have a full path match option, rather than this stop iteration stuff.

Reply:
Maybe we should return an error if the user passes a glob? Passing a single path without `*` to getTarConditional should only match one path.  Maybe have a separate getTarSingle function which wraps getTarConditional and panics if the callback is called more than once.

Reply:
Looking at this some more. I think the right thing to do is error if the path/glob is ambiguous.